performance_manager/lib/l2_dwell_travel_times.py

```diff
@@ -0,0 +1,165 @@
+import sqlalchemy as sa
+
+from .logging_utils import ProcessLogger
+from .postgres_utils import DatabaseManager
+from .postgres_schema import (
+    VehiclePositionEvents,
+    FullTripEvents,
+    DwellTimes,
+    TravelTimes,
+)
+
+
+def dwell_times(db_manager: DatabaseManager) -> None:
+    """
+    perform update on dwell times table
+
+    executed as delete and insert
+    """
+    process_logger = ProcessLogger("update l2 dwell times")
+    process_logger.log_start()
+
+    # base select for insert of records dwell times
+    # dwell time = timestamp.end - timestamp.start for stop portion of trip events
+    insert_select = sa.select(
+        VehiclePositionEvents.pk_id,
+        (
+            VehiclePositionEvents.timestamp_end
+            - VehiclePositionEvents.timestamp_start
+        ).label("dwell_time_seconds"),
+    ).join(
+        FullTripEvents,
+        VehiclePositionEvents.pk_id == FullTripEvents.fk_vp_stopped_event,
+    )
+
+    # pull last created_on timestamp from dwell time table
+    max_create_travel_query = sa.select(sa.func.max(DwellTimes.created_on))
```

Comment:
how does this scale as `DwellTimes` blows up?

Reply:
All of these selects are based on timestamp fields which should be very well optimized in postgres. So I don't think we'll have any issues with the scaling of these operations.